src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -123,6 +126,8 @@ trait Blockchain {
 
   def removeBlock(hash: ByteString): Unit
 
+  def removeBlock(blockNumber: BigInt): Unit = getHashByBlockNumber(blockNumber).foreach(removeBlock)
```

Comment:
I don't think this removes the entry from `blockNumberMappingStorage`. Should it?

Reply:
You are right! :+1:
I will add the removal of that entry to `def removeBlock(hash: ByteString): Unit` implementation.